Hero.cpp

```diff
@@ -28,16 +29,19 @@ void Hero::xpGain(int gain) {
 }
 
 void Hero::tamad(Monster* a) {
-    int gain = dmg;
-	if (dmg > a->hp) gain = a->hp;
-	a->hp = a->hp - this->dmg;
-	xpGain(gain);
+    if (this->dmg > a->def)
+    {
+        int gain = this->dmg - a->def;
+        if ((this->dmg - a->def) > a->hp) gain = a->hp;
+        a->hp = a->hp - (this->dmg - a->def);
+        xpGain(gain);
+    }

```

Comment:
- osztályon belül kell this?
- az if-be this->dmg - a->def helyére nem lehet a gain-t írni?
- 36-os sorba = helyett nem jobb a -= ?

Reply:
föltűnt, hogy itt lehet negatív a->hp értéke, viszont a Monster::tamad() nál nincs megengedve. Nem akad így össze valami? Ha működik, maradhat, csak gondoltam szólok.

Reply:
- negatív hpval nem lép bele ebbe a függvénybe, úgyhogy nem okoz problémát,
- az if-es résznél nele lehetne írni, mert valóban egyenlő a két érték viszont logikailag nem lenne jó, mert nem azt akarom tesztelni, hogy mennyi xp-t kap hanem hogy a damage és a defense hogy viszonyul egymáshoz
- a 36-os sornál: csak szemantikai eltérés van, futásban nincs különbség